specification/e-referrals-service-api.yaml

```diff
@@ -3303,7 +3303,89 @@ paths:
         - bearerAuth: []
       description: |
         ## Overview
-         TBD
+        Use this endpoint to cancel a referral and the current appointment, if there is one. Cancelling a referral is a one-way process and results in an end-state. No further progression of care can occur once this action has been performed.
+
+        A reason for cancelling an appointment must be provided. A comment on the cancellation is conditionally mandatory depending on the reason selected. See pre-requisites and related referrals for more information.
+
+        You can check if you can cancel a referral by calling [[HYPERLINK_A029]].
+
+        e-RS does not allow a referral to be cancelled if there is a current appointment. Therefore, if there is a current appointment associated with the referral, this *must* be cancelled in advance, or at the same time.
+
+        In order to cancel (only) an appointment you must use [[HYPERLINK_A034]].
+
+        ## Supported security patterns
+        - Healthcare worker, user-restricted access
+
+        ## Pre-requisites
+        In order to use this endpoint you must be an authenticated e-RS user and use one of the following e-RS roles:
+        - `REFERRING_CLINICIAN`
+        - `REFERRING_CLINICIAN_ADMIN`
+
+        You must use [[HYPERLINK_A004]] to retrieve a list of reference data, which detail the reason the cancellation is being performed. You can only use reference data from the set `REFERRAL-CANCELLATION-REASON`.
+
+
+        To find out which reference data from this set can be used, the following filters can be used depending on the cancellation scenario:
+
+        ***1. I want to cancel the referral and the current appointment***
+
+        * Obtain reference data from `REFERRAL-CANCELLATION-REASON` and filter:
+
+          * `referrerUsage` must be `CANCEL_REFERRAL`, and
+
+          * `businessFunction` must be `REFERRING_CLINICIAN` or `REFERRING_CLINICIAN_ADMIN` (depending which authenticated user will be performing the action), and
+
+          * `canCancelAppointment` must be true, and
+
+          * `effectiveFrom` must be in the past, and 
+
+          * if provided, `effectiveTo` must **not** be in the past
+
+
+        ***2. I want to cancel a referral, for which there is no current appointment***
+
+        * Obtain reference data from `REFERRAL-CANCELLATION-REASON` and filter:
+
+          * `referrerUsage` must be `CANCEL_REFERRAL`, and
+
+          * `businessFunction` must be `REFERRING_CLINICIAN` or `REFERRING_CLINICIAN_ADMIN`   *(depending which authenticated user will be performing the action), and
+
+          * `canCancelAppointment` could be true or false, and
+
+          * `effectiveFrom` must be in the past, and
```

Comment:
For this endpoint, is there something that states that the effectiveFrom should Not be null?

Some refdata entries currently do not have an effectiveFrom date, although these entries aren't retrievable for this endpoint.

Reply:
BAs decided to leave as is, because it has no functional impact on the endpoint. See https://nhsd-jira.digital.nhs.uk/browse/ERSSUP-58954